### ISSUE TITLE
Default Keybindings Not applied?

### ISSUE THREAD

**Issue by WimpieRatte** [blay09/DefaultKeyConfig#72]
### Minecraft Version

1.20.x

### Mod Loader

Forge

### Mod Loader Version

47.2.1

### Mod Version

18.0.1

### Balm Version

7.1.4

### Describe the Issue

It seems that the keybinds that I specified are not being applied when the person installs a new instance of the modpack.
Other settings, like video, shaders and resource packs, all seem to work, at least.

It's just a bit frustrating to know that new installs will have so many shortcut conflicts.

I'm not sure which details I can share to help troubleshoot this.
This is the modpack: https://modrinth.com/modpack/colonies-and-technologies
If you install a new instance, you'll notice that even though the "config\defaultoptions\keybindings.txt" has crawl bound to "caps lock", when the game loads, the crawl is not bound to caps lock :(.

> key_key.crawlondemand.crawl:key.keyboard.caps.lock:NONE

This is especially "urgh" for super common shortcut keys, like "r".
I don't want to override existing keybinds by publishing the options.txt file each time... so I'm really hoping this is a bug that can be confirmed and fixed :D.

**Comment by natephillips801** [comment]
This is happening to me too. I created a fresh modpack and tested on forge 47.1.0 and 47.2.0. After modifying some keybinds, saving them, verifying they are correct in the configs and then wiping and regenerating the options.txt the configs do not update the options.txt.

**Comment by BlayTheNinth** [comment]
Are you deleting `knownkeys.txt` too when testing? If the keys are in knownkeys.txt Default Options will only set the Default of the keybinding, but not change the actual bound key.

I can not reproduce any issue using the Colonies and Technologies pack. If I delete the shipped options.txt and start it up fresh, my crawl is correctly bound to caps lock.

**Comment by mahjerion** [comment]
I can also confirm this issue during development of Craft to Exile 2. Oddly, it works for some users and some users don't have the custom default keybinds on install.

EDIT:

More context:
MC: 1.20.1
Forge: 47.2.1

**Comment by WimpieRatte** [comment]
Ok, so should I publish the knownkeys.txt, but not the options.txt?
Or do you mean I should put knownkeys.txt into the "defaultoptions\extra" folder?

**Comment by BlayTheNinth** [comment]
Do not ship `knownkeys.txt` and `options.txt` and delete **both** when testing if they work.

**Comment by WimpieRatte** [comment]
My process was to only export these folders:
		config
		mods
		resourcepacks
		shaderpacks

If someone already had a client installed and updated it, then his settings were kept.
The issue was when someone installed a new instance. The mods' usual defaults were assigned instead of the keybinds that I wanted for the pack.

I'll publish my new version without the options.txt, because I cannot remember when I reverted to rather include the options.txt.
After the update, if a new installation with the new version doesn't have the correct keybinds, I'll load a new version with options.txt again, but can then tell you exactly which colonies and technologies version will demonstrate the issue.
Sorry for not mentioning the exact versioning when I logged the ticket :(.

**Comment by WimpieRatte** [comment]
Everything seems to now work fine on C&T version 22.
Maybe a different mod prevented yours from working properly and then got updated to no longer interrupt... I don't know.
If I ever get this issue again and I can recreate it like I could back then, I'll message here again.
Thanks for the inputs and trying to investigate!

**Comment by WimpieRatte** [comment]
Closing ticket - unless someone can specify how we can recreate the issue :).

**Comment by WimpieRatte** [comment]
Ok. Found something that might be re-creatable.
On my client that I "updated" to 22 (not a new instance), everything was reset to the mod defaults keybinds.
HOWEVER, when I went to the key bindings and clicked "reset all", it set it according to how I originally set it up on the modpack publication.

So it was super quick to fix client-side, but the issue is still there.

Does this help?

**Comment by mahjerion** [comment]
Thanks for the workaround @WimpieRatte . Hopefully it gets resolved but at least we have this

**Comment by Proxwian** [comment]
I can confirm that keybinds do not apply on my modpack.

How to recreate issue:
1) Install TerraFirmaAdventure modpack
2) See keybinds
3) Now reset it to default - it becomes now as defaultSettings should do them
4) Close the game, delete options.txt
5) Start game, see keybinds settings - they become wrong again (not from defaultOptions)

**Comment by BlayTheNinth** [comment]
You need to delete `knownkeys.txt` as well.

**Comment by Proxwian** [comment]
It also doesn't help as players installs modpack without ```knownkeys.txt``` file and still reports the issue

**Comment by PepGames** [comment]
> It also doesn't help as players installs modpack without `knownkeys.txt` file and still reports the issue

Yea when I export config, mods, resourcepacks, and shaders, then launch that exported version my defaults are not applied whatsoever and the player has to manually go through and reset every single keybind for them to be what I wanted for the pack.

**Comment by Fyoncle** [comment]
I also have this problem, knownkeys.txt automatically getting created on modpack's players even tho we dont include it after the first boot they do, that's strange.

**Comment by Jadan1213** [comment]
i also ran into an issue related to this. our keybinds would constantly reset. they'd be fine on first install, but subsequent starts would have all the keys reset to mod defaults. 

My issue that i discovered was that if the game crashed/didn't close properly, the keys would be reset the next startup. My pack's crashing on exit issue was caused by seamless loading screen. I removed that and "so far" <crosses fingers> i haven't had them reset post install.
